### PR TITLE
Handle context=None in convert_unit when input/output GHG species match

### PR DESCRIPTION
**Pull request by khaeru**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- [x] Description in RELEASE_NOTES.md Added

# Description of PR

Adds tests for #368 and adjusts to IAMconsortium/units#21, which contains the fix.

### CODE REVIEW

**Comment by khaeru** [comment]
@danielhuppmann—I haven't introduced any CI changes here to run against the branch for IAMconsortium/units#21, but the combination of the two works on my system.

I would suggest:
1. Check that the combination also works for you.
2. Approve & merge the iam-units PR.
3. I'll make a quick release of iam-units.
4. Let CI re-run here, confirm that checks pass.
5. Approve and merge this PR.

The tests have been expanded, but if there is any other untested but expected behaviour, this would be a good time to add it—please add to the branch.

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -2,13 +2,14 @@
 
 ## Individual Updates
 
-- [#361](https://github.com/IAMconsortium/pyam/pull/361) iam-units refactored from a Git submodule to a Python dependency of pyam.
+- [#361](https://github.com/IAMconsortium/pyam/pull/361),
+  [#369](https://github.com/IAMconsortium/pyam/pull/369) iam-units refactored from a Git submodule to a Python dependency of pyam.
```
Comment (danielhuppmann) [inline RELEASE_NOTES.md:6]: please order in descending order and add description like "bugfix to allow same-species mass conversion"

Comment (khaeru) [reply RELEASE_NOTES.md]: Done!

**Comment by khaeru** [comment]
Per discussion in Slack last week:
- IAMconsortium/units#21 expanded to ensure that *different symbols for the same GHG species*, e.g. `CO2_eq` and `CO2e`, can be converted to one another, without supplying a metric name.
- This PR expanded to handle *pyam's supported lower-case aliases* for chemical symbols, e.g. using `co2` to mean CO₂.
- Tests are added for both.

> 1. Check that the combination also works for you.
> 2. Approve & merge the iam-units PR.
> 3. I'll make a quick release of iam-units.
> 4. Let CI re-run here, confirm that checks pass.

These steps are now complete.

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -2,13 +2,14 @@
 
 ## Individual Updates
 
+- [#369](https://github.com/IAMconsortium/pyam/pull/369) `convert_units()`: support GWP conversion of same GHG species without context, lower-case aliases for species symbols.
```
Comment (danielhuppmann) [inline RELEASE_NOTES.md:5]:
```suggestion
- [#369](https://github.com/IAMconsortium/pyam/pull/369) `convert_unit()` supports GWP conversion of same GHG species without context, lower-case aliases for species symbols.
```

File: pyam/core.py [hunk 3]
```diff
@@ -766,7 +766,7 @@ def rename(self, mapping=None, inplace=False, append=False,
 
     def convert_unit(self, current, to=None, factor=None, registry=None,
                      context=None, inplace=False):
-        """Convert all data having *current* units to new units.
+        r"""Convert all data having *current* units to new units.
```
Comment (danielhuppmann) [inline pyam/core.py:769]: is the `r` on purpose?

Comment (khaeru) [reply pyam/core.py:769]: Yes: I built the docs, and saw that `'gwp_'` further down was being interpreted as a ReST link (trailing underscore) when I didn't intend that. So, I added `\` to escape the underscore; but then my Python linter was complaining that that was a special character. `r` makes it a ["raw string"](https://docs.python.org/3/reference/lexical_analysis.html#index-19) in which `\` is a literal character.

**Comment by danielhuppmann** [comment]
resolved the merge conflict

**Pull request merged**